luaui/Widgets/cmd_nano_range_check.lua

```diff
@@ -0,0 +1,277 @@
+-- This is faster then synch reading from the engine; jumping.
+-- 1.4e changed to use 2D distance x and z instead of 3D, as it seems that nanos work in 2D cylinders.
+local function calculateDistance2D(x1, y1, x2, y2)
+    return math.sqrt((x1 - x2)^2 + (y1 - y2)^2)
+end
+
+-- LRU caching of model radius, so we don't have to get it every time
+local function getUnitRadius(uID)
+    local radius = radius_cache:get(uID)
+    if not radius then
+        radius = Spring.GetUnitDefDimensions(Spring.GetUnitDefID(uID)).radius
+        radius_cache:put(uID, radius)
+    end
+    return radius
+end
+
+-- Checks if a command is valid
+local function isValidCommandID(commandID)
+    return commandID == CMD.REPAIR or commandID == CMD.GUARD or commandID == CMD.RECLAIM or commandID == CMD.ATTACK
+end
+-- !SECTION Helper functions
+
+-- SECTION Main functions
+local function checkTurretRange(uID)
+    local x, y, z = Spring.GetUnitPosition(uID)
+    local build_distance = Spring.GetUnitEffectiveBuildRange(uID, nil) -- Same as UnitDef.buildDistance; Not ambiguous
+    local is_changed = false
+    local is_first_cmd = true
+    local cmds = Spring.GetUnitCommands(uID, -1)
+    local new_cmds = {}
+    for i = #cmds, 1, -1  do
+        local cmd = cmds[i]
+        if isValidCommandID(cmd.id) then
+            local tuID = cmd.params[1]
+            local tx, ty, tz = Spring.GetUnitPosition(tuID)
+            if tx == nil then break end
+            local distance = calculateDistance2D(x, z, tx, tz)
+            local radius = getUnitRadius(tuID)
+            if distance < build_distance + radius then -- BP uses build_distance + radius (sphereical shape of model)
+                cmd.options.shift = not is_first_cmd
+                is_first_cmd = false
+                table.insert(new_cmds, {cmd.id, cmd.params, cmd.options}) -- building cmdArr
+            else
+                is_changed = true
+            end
+        end
+    end
+    if is_changed then
+        if #new_cmds > 0 then
+            table.insert(new_orders, {uID, new_cmds}) -- schedule for new order
+        else
+            table.insert(to_be_cleared, uID) -- schedule for clear
+        end
+    end
+end
+
+-- Applies all stop orders to units in the `to_be_cleared` stack.
+local function applyStopOrders()
+    if #to_be_cleared > 0 then
+        Spring.GiveOrderToUnitArray(to_be_cleared, CMD.STOP, {}, {})
+        command_budget = command_budget - #to_be_cleared
+        to_be_cleared = {}
+    end
+end
+
+-- Processes new orders that have been scheduled for execution in the `new_orders` stack.
+local function processNewOrders()
+    if #new_orders > 0 and #to_be_cleared == 0 then
+        for i = #new_orders, 1, -1 do
+            if command_budget <= 0 then break end
+            Spring.GiveOrderArrayToUnit(new_orders[i][1], new_orders[i][2])
+            table.remove(new_orders, i) -- pop
+            command_budget = command_budget - 1
+        end
+    end
+end
```

Comment:
```suggestion
local function processNewOrders()
    if #new_orders == 0 or #to_be_cleared > 0 then
        return
    end

    for i = #new_orders, 1, -1 do
        if command_budget <= 0 then
            break
        end
        
        Spring.GiveOrderArrayToUnit(new_orders[i][1], new_orders[i][2])
        table.remove(new_orders, i) -- pop
        command_budget = command_budget - 1
    end
end
```

prefer non-nested logic